pystac_client/cli.py

```diff
@@ -124,7 +124,7 @@ def collections(
             else:
                 raise KeyError("'matched' is not supported for this catalog")
         else:
-            collections_dicts = [c.to_dict() for c in result.collections()]
+            collections_dicts = list(result.collections_as_dicts())
```

Comment:
This is an untested change, but seems reasonable to _not_ have to parse all the dictionaries as valid-ish collections and then turn them right back into dicts.